### ISSUE TITLE
Filesystem gets unmounted and re-assigned to different device after accessing S.M.A.R.T screen

### ISSUE THREAD

**Issue by jefbam** [openmediavault/openmediavault#901]
After clicking on S.M.A.R.T icon, the serial number for a USB-attached drive is changed and the filesystem is unmounted and re-assigned to a different device name.

Equipment:
- OWC Mercury Elite Pro Dual USB enclosure with two (2) 2TB drives configured as RAID1 in onboard-enclosure. Connected to R-PI via USB 3.0 port
- R-PI 4 4GB
- OMV 5.5.19-1

Procedure:
Connect and power-on the enclosure to R-PI

Observed results:
Under Storage, click Disks. Screen appears:
Device=/dev/sdb, Model=Pro Dual-R1, Serial Number=RANDOM 9FC090397067, Vendor=Elite, Capacity=1.82TiB

Under Storage, click File Systems. Sreen appears:
Device=/dev/sdb1, Label=OWC, Filesystem=ntfs, Total=n/a, Available=n/a, Used=n/a, Mounted=No, Referenced=No, Status=Online

Select sdb1 and click Mount. Click Apply to save changes.

Under Storage, click File Systems. Screen appears:
Device=/dev/sdb1, Label=OWC, Filesystem=ntfs, Total=1.82 TiB, Available=617.27 GiB, Used=1.21 TiB, Mounted=Yes, Referenced=No, Status=Online

Shared Folders can be setup successfully and CIFS shares can be setup successfully and file sharing works well.

Under Storage, click S.M.A.R.T.

Error occurs:

Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; blockdev --getsize64 '/dev/sdb' 2>&1' with exit code '1': blockdev: cannot open /dev/sdb: No such file or directory

Error #0:
OMV\ExecException: Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; blockdev --getsize64 '/dev/sdb' 2>&1' with exit code '1': blockdev: cannot open /dev/sdb: No such file or directory in /usr/share/php/openmediavault/system/process.inc:182
Stack trace:
#0 /usr/share/php/openmediavault/system/blockdevice.inc(258): OMV\System\Process->execute(Array)
#1 /usr/share/openmediavault/engined/rpc/smart.inc(99): OMV\System\BlockDevice->getSize()
#2 /usr/share/php/openmediavault/rpc/serviceabstract.inc(635): Engined\Rpc\Smart->Engined\Rpc\{closure}()
#3 /usr/share/openmediavault/engined/rpc/smart.inc(107): OMV\Rpc\ServiceAbstract->asyncProc(Object(Closure))
#4 [internal function]: Engined\Rpc\Smart->enumerateDevices(NULL, Array)
#5 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#6 /usr/share/openmediavault/engined/rpc/smart.inc(174): OMV\Rpc\ServiceAbstract->callMethod('enumerateDevice...', NULL, Array)
#7 [internal function]: Engined\Rpc\Smart->getList(Array, Array)
#8 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#9 /usr/share/php/openmediavault/rpc/serviceabstract.inc(149): OMV\Rpc\ServiceAbstract->callMethod('getList', Array, Array)
#10 /usr/share/php/openmediavault/rpc/serviceabstract.inc(588): OMV\Rpc\ServiceAbstract->OMV\Rpc\{closure}('/tmp/bgstatusaR...', '/tmp/bgoutputVE...')
#11 /usr/share/php/openmediavault/rpc/serviceabstract.inc(159): OMV\Rpc\ServiceAbstract->execBgProc(Object(Closure))
#12 /usr/share/openmediavault/engined/rpc/smart.inc(209): OMV\Rpc\ServiceAbstract->callMethodBg('getList', Array, Array)
#13 [internal function]: Engined\Rpc\Smart->getListBg(Array, Array)
#14 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#15 /usr/share/php/openmediavault/rpc/rpc.inc(86): OMV\Rpc\ServiceAbstract->callMethod('getListBg', Array, Array)
#16 /usr/sbin/omv-engined(537): OMV\Rpc\Rpc::call('Smart', 'getListBg', Array, Array, 1)
#17 {main}

Click OK

Under Storage, click Disks. Screen appears:

Device=/dev/sdb, Model=Pro Dual-R1, Serial Number=RANDOM E7AD758C4668, Vendor=Elite, Capacity=1.82TiB

(Note: the Serial Number has changed)

Under Storage, click File Systems. Screen appears:

Device=/dev/sdc1, Label=OWC, Filesystem=ntfs, Total=n/a, Available=n/a, Used=n/a, Mounted=No, Referenced=No, Status=Online

(Note: filesystem is unmounted, and has changed from sdb1 to sdc1)

Select sdc1 and click Mount.

At this point, if any shares have been setup, they no longer work. All shared folders must be deleted and recreated to work correctly.

Expected Results:

Device serial number not to change when selecting S.M.A.R.T screen
No errors when selecting S.M.A.R.T screen
Selecting S.M.A.R.T screen does not initiate re-mounting filesystem to different device name

**Additional info:

Running the following commands yield the same behavior. The command output is included below. If I need to enter this defect info in a Raspbian forum or something please advise. Note that I changed the actual serial number to X's.

$ sudo hdparm -I /dev/sd? | grep 'Serial\ Number'

The running kernel lacks CONFIG_IDE_TASK_IOCTL support for this device.
READ_LOG_EXT(0,0) failed: Invalid argument
        Serial Number:      XXXXXXXXXXXXXXXXXXXX

$ sudo hdparm -I /dev/sdb1 | grep 'Serial\ Number'
 HDIO_DRIVE_CMD(identify) failed: Invalid argument

(Note: running these commands unmounted and reassigned the drive to sdc)

$ sudo hdparm -I /dev/sdc

/dev/sdc:

ATA device, with non-removable media
        Model Number:       Elite Pro Dual-R1
        Serial Number:      XXXXXXXXXXXXXXXXXXXX
        Firmware Revision:  13521050
Standards:
        Supported: 7 6 5 4
        Likely used: 7
Configuration:
        Logical         max     current
        cylinders       16383   16383
        heads           16      16
        sectors/track   63      63
        --
        CHS current addressable sectors:    16514064
        LBA    user addressable sectors:   268435455
        LBA48  user addressable sectors:  3906928640
        Logical  Sector size:                   512 bytes
        Physical Sector size:                   512 bytes
        device size with M = 1024*1024:     1907680 MBytes
        device size with M = 1000*1000:     2000347 MBytes (2000 GB)
        cache/buffer size  = unknown
Capabilities:
        LBA, IORDY(can be disabled)
        Standby timer values: spec'd by Vendor, no device specific minimum
        R/W multiple sector transfer: Max = 1   Current = 1
        Advanced power management level: disabled
        DMA: mdma0 mdma1 mdma2 udma0 udma1 udma2 udma3 udma4 udma5 *udma6
             Cycle time: min=120ns recommended=120ns
        PIO: pio0 pio1 pio2 pio3 pio4
             Cycle time: no flow control=240ns  IORDY flow control=120ns
Commands/features:
        Enabled Supported:
           *    SMART feature set
                Security Mode feature set
           *    Power Management feature set
           *    Write cache
                Look-ahead
           *    Host Protected Area feature set
           *    WRITE_BUFFER command
           *    READ_BUFFER command
           *    NOP cmd
                Advanced Power Management feature set
                SET_MAX security extension
           *    48-bit Address feature set
           *    Mandatory FLUSH_CACHE
           *    FLUSH_CACHE_EXT
           *    WRITE_{DMA|MULTIPLE}_FUA_EXT
           *    Gen1 signaling speed (1.5Gb/s)
           *    Gen2 signaling speed (3.0Gb/s)
           *    Gen3 signaling speed (6.0Gb/s)
           *    Software settings preservation
Security:
        Master password revision code = 65534
                supported
        not     enabled
        not     locked
        not     frozen
        not     expired: security count
        not     supported: enhanced erase
        more than 508min for SECURITY ERASE UNIT.
Checksum: correct

**Comment by votdev** [comment]
The issue is caused by your USB drive enclosure. It is surely using a JMicron controller. Changing the serial ID during runtime seems to be a new behavior, i never heard about that before.

There is another issue caused by JMicron: https://github.com/openmediavault/openmediavault/issues/896

**Comment by votdev** [comment]
And yes, it is based on JMicron.

See https://dittools.eu/products/owc-mercury-elite-pro-dual-performance-raid-enclosure-usb
```
Interface (Chipset)

    (1) USB 3.1 Gen 1 (JMicron JMS562)
    (1) eSATA (JMicron JMS562)
    Interface Controller: ASMedia ASM1456B
```

See my comment about that manufacturer [here](https://github.com/openmediavault/openmediavault/issues/896#issuecomment-751511345).

**Comment by votdev** [comment]
Please run the following commands and post the output here for protocol and research issues:
```
# udevadm info --query=property --name=/dev/sdb
# udevadm info --query=property --name=/dev/sdc
# blkid
```
You will realize that the same serial ID is reported by every device. This serial ID is used to generate a UNIQUE device file by the kernel that is used to mount the filesystem. The problem is, your enclosure reports a normally unique serial ID for multiple devices which will finally confuse the kernel and userland tools.

**Comment by jefbam** [comment]
Thanks for the info. Incidentally, please let me know if you know of any 2-bay RAID USB enclosures that work well with R-PI/OMV. Thanks again.

The command results are:

(Note: the device name changed from sdc to sdd since the original posting)

$ udevadm info --query=property --name=/dev/sdd
DEVPATH=/devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb2/2-1/2-1:1.0/host3/target3:0:0/3:0:0:0/block/sdd
DEVNAME=/dev/sdd
DEVTYPE=disk
MAJOR=8
MINOR=48
SUBSYSTEM=block
USEC_INITIALIZED=8654497300
ID_VENDOR=Elite
ID_VENDOR_ENC=Elite\x20\x20\x20
ID_VENDOR_ID=1e91
ID_MODEL=Pro_Dual-R1
ID_MODEL_ENC=Pro\x20Dual-R1\x20\x20\x20\x20\x20
ID_MODEL_ID=a3a6
ID_REVISION=0105
ID_SERIAL=Elite_Pro_Dual-R1_RANDOM__D61B6A084014-0:0
ID_SERIAL_SHORT=RANDOM__D61B6A084014
ID_TYPE=disk
ID_INSTANCE=0:0
ID_BUS=usb
ID_USB_INTERFACES=:080650:080662:
ID_USB_INTERFACE_NUM=00
ID_USB_DRIVER=usb-storage
ID_PATH=platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1:1.0-scsi-0:0:0:0
ID_PATH_TAG=platform-fd500000_pcie-pci-0000_01_00_0-usb-0_1_1_0-scsi-0_0_0_0
ID_PART_TABLE_UUID=93568ccc
ID_PART_TABLE_TYPE=dos
DEVLINKS=/dev/disk/by-id/usb-Elite_Pro_Dual-R1_RANDOM__D61B6A084014-0:0 /dev/disk/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1:1.0-scsi-0:0:0:0
TAGS=:systemd:

$ blkid
/dev/mmcblk0p1: LABEL_FATBOOT="boot" LABEL="boot" UUID="EBBA-157F" TYPE="vfat" PARTUUID="d9bfdc03-01"
/dev/mmcblk0p2: LABEL="rootfs" UUID="b3ce35cd-ade9-4755-a4bb-1571e37fc1b9" TYPE="ext4" PARTUUID="d9bfdc03-02"
/dev/sdd1: LABEL="OWC" UUID="3A3A95FB3A95B47B" TYPE="ntfs" PARTUUID="93568ccc-01"

**Comment by votdev** [comment]
openmediavault `5.5.20` has been released, please give it a try and report if the issue has been fixed.

**Comment by votdev** [comment]
> Device=/dev/sdb, Model=Pro Dual-R1, Serial Number=RANDOM 9FC090397067, Vendor=Elite, Capacity=1.82TiB
> 
> ID_SERIAL_SHORT=RANDOM__D61B6A084014

Wow, this is amazing. This chipset really rolls the serial number.

**Comment by jefbam** [comment]
Ok, thanks. I will try the new ver.

Incidentally, if I do sudo hdparm -I /dev/sdd, it does always report the actual serial number (I've changed the last 10 positions):

ATA device, with non-removable media
        Model Number:       Elite Pro Dual-R1
        Serial Number:      XB0AH3E6LMXXXXXXXXXX
        Firmware Revision:  13521050

**Comment by jefbam** [comment]
Ok, I've installed 5.5.20. 

The very first time SMART is clicked, no errors occur, the fs is no longer re-mounted, everything is great.

If the user clicks off of SMART to another screen (e.g., Disks, Filesystem, etc.) and then clicks SMART again, the error occurs; it is only the second and subsequent clicks of the SMART icon that now cause the error conditions.

**Comment by votdev** [comment]
> Ok, I've installed 5.5.20.
> 
> The very first time SMART is clicked, no errors occur, the fs is no longer re-mounted, everything is great.

Great to hear

> If the user clicks off of SMART to another screen (e.g., Disks, Filesystem, etc.) and then clicks SMART again, the error occurs; it is only the second and subsequent clicks of the SMART icon that now cause the error conditions.

Sorry, what error appears?

I remember a thread in the forum where someone reported a similar behavior that the system behaves strange after SMART values were requested. I think the user could fix it by installing a new firmware for his enclosure. Sadly i can not find the post anymore.

**Comment by jefbam** [comment]
Sure, the error message and unmounting of the filesystem is the same behavior that used to happen on the first click of SMART icon--before the 5.5.20 update. Now it only happens on the second and subsequent clicks of SMART icon:

Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; blockdev --getsize64 '/dev/sda' 2>&1' with exit code '1': blockdev: cannot open /dev/sda: No such device or address

Error #0:
OMV\ExecException: Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; blockdev --getsize64 '/dev/sda' 2>&1' with exit code '1': blockdev: cannot open /dev/sda: No such device or address in /usr/share/php/openmediavault/system/process.inc:182
Stack trace:
#0 /usr/share/php/openmediavault/system/blockdevice.inc(258): OMV\System\Process->execute(Array)
#1 /usr/share/openmediavault/engined/rpc/smart.inc(99): OMV\System\BlockDevice->getSize()
#2 /usr/share/php/openmediavault/rpc/serviceabstract.inc(635): Engined\Rpc\Smart->Engined\Rpc\{closure}()
#3 /usr/share/openmediavault/engined/rpc/smart.inc(107): OMV\Rpc\ServiceAbstract->asyncProc(Object(Closure))
#4 [internal function]: Engined\Rpc\Smart->enumerateDevices(NULL, Array)
#5 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#6 /usr/share/openmediavault/engined/rpc/smart.inc(174): OMV\Rpc\ServiceAbstract->callMethod('enumerateDevice...', NULL, Array)
#7 [internal function]: Engined\Rpc\Smart->getList(Array, Array)
#8 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#9 /usr/share/php/openmediavault/rpc/serviceabstract.inc(149): OMV\Rpc\ServiceAbstract->callMethod('getList', Array, Array)
#10 /usr/share/php/openmediavault/rpc/serviceabstract.inc(588): OMV\Rpc\ServiceAbstract->OMV\Rpc\{closure}('/tmp/bgstatusSQ...', '/tmp/bgoutputw2...')
#11 /usr/share/php/openmediavault/rpc/serviceabstract.inc(159): OMV\Rpc\ServiceAbstract->execBgProc(Object(Closure))
#12 /usr/share/openmediavault/engined/rpc/smart.inc(209): OMV\Rpc\ServiceAbstract->callMethodBg('getList', Array, Array)
#13 [internal function]: Engined\Rpc\Smart->getListBg(Array, Array)
#14 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#15 /usr/share/php/openmediavault/rpc/rpc.inc(86): OMV\Rpc\ServiceAbstract->callMethod('getListBg', Array, Array)
#16 /usr/sbin/omv-engined(537): OMV\Rpc\Rpc::call('Smart', 'getListBg', Array, Array, 1)
#17 {main}

I was on a call with the vendor this morning and very unfortunately they do not have a firmware update for the enclosure; they only provide Windows and Mac drivers. I'll try again to find the posting regarding updating the firmware.

Thanks for your efforts.

**Comment by jefbam** [comment]
Adding information here if helpful:

As noted earlier in this issue, this command:
$ sudo hdparm -I /dev/sda1 | grep 'Serial\ Number'
 
Results in:
HDIO_DRIVE_CMD(identify) failed: Invalid argument

The filesystem is then unexpectedly unmounted, and a new "RANDOM" serial number is reported in the Disks screen of OMV; the same behavior as when the SMART screen is clicked.

However, if I connect this USB enclosure to a R-PI 3 (Raspbian GNU/Linux 9 stretch) and issue the same command, the result is:

       Serial Number:      XB0AH3E6LMXXXXXXXXXX

(I've replaced the last positions with X's)

And the filesystem remains mounted with no problems.

**Comment by jefbam** [comment]
I've discovered the cause of the errors and unexpected unmounting of the filesystem when using the hdparm command or clicking the SMART screen in the UI. I had previously made an entry in the /boot/cmdline.txt file: usb-storage.quirks=1e91:a3a6:u due to slow performance of writes to the drive. I've since concluded that the performance is due to the volume is ntfs and will convert it to ext4 in the future.

In any case, I removed the entry in the cmdline.txt and I have had no problems since. Thanks for the help with this issue.
